### PR TITLE
Fixes #4290 - Automatically creating tokens only when a token isn't set

### PR DESCRIPTION
**Pull request by gevans**
Please see http://dev.kohanaframework.org/issues/4290

### CODE REVIEW

**Comment by gevans** [comment]
Also, not 100% sure this is something that should be applied to `3.2/develop` or held off until there's a 3.3 branch.

**Comment by gevans** [comment]
Nice.

```
22:12:48  There was 1 failure:
22:12:48  
22:12:48  1) Kohana_Cache_FileTest::test_set_get with data set #6 (array('object', stdClass, 0, false, 'object', NULL), stdClass)
22:12:48  Failed asserting that <null> is equal to 
22:12:48  stdClass Object
22:12:48  (
22:12:48      [foo] => foo
22:12:48      [bar] => bar
22:12:48  )
```

**Comment by kiall** [comment]
Humm .. looks like a bug in the cache tests triggered this to fail...

!build

File: classes/model/auth/user/token.php [hunk 1]
```diff
@@ -49,9 +49,19 @@ public function delete_expired()
...
+	 * the database.
+	 *
+	 * @param   Validation  $validation  Validation object
+	 * @return  ORM
+	 */
 	public function create(Validation $validation = NULL)
 	{
-		$this->token = $this->create_token();
+		if ($this->token === NULL OR ORM::factory('user_token', array('token' => $token))->loaded())
+		{
```
Comment (kemo) [inline classes/model/auth/user/token.php:62]: I don't see `$token` defined anywhere?

**Comment by kemo** [comment]
Targeted against the wrong branch + undefined variable `$token` used in `ORM::factory('user_token', array('token' => $token))->loaded()`. Closing

**Pull request closed without merging**